crates/astria-conductor/src/executor.rs

```diff
@@ -221,9 +250,15 @@ impl<C: ExecutionClient> Executor<C> {
 
         let response = self
             .execution_rpc_client
-            .call_do_block(prev_block_hash, txs, Some(timestamp))
+            .call_do_block(prev_execution_block_hash, txs, Some(timestamp))
             .await?;
         self.execution_state = response.block_hash.clone();
+        self.most_recently_executed_block_hash =
+            if let Some(parent_block) = block.header.last_block_id {
+                parent_block.hash
+            } else {
+                Hash::default()
+            };
```

Comment:
shouldn't this be `self.most_recently_executed_block_hash = block.block_hash`?

Reply:
yes.